src/mmg2d/analys_2d.c

```diff
@@ -524,8 +524,8 @@ int MMG2D_regnor(MMG5_pMesh mesh) {
   maxit = 10;
   res0 = 0.0;
   nn = 0;
-  lm1 = 0.4;
-  lm2 = 0.399;
+  lm1 = mesh->info.lxreg;
+  lm2 = lm1;
```

Comment:
I guess that the slight difference between the `lm1` and `lm2` values is here to avoid cases of oscillations during regularization... Is it possible to check that (by reading the code because it may be hard and quite random to reproduce oscillations)?

Reply:
To avoid oscillations, i changed the computation of _lm2_ so that it is always slightly lower than _lm1_ .